### DISCUSSION TITLE
Struggling to Add Text Onto Image

### DISCUSSION THREAD

**Discussion by Trademark8901** [ImageMagick/ImageMagick#2685] [Help] [answered]
Hello,

I am trying to create a process cmd in Windows that will grab the EXIF date and time data from a picture and stamp it onto the picture itself. I am currently stuck on the process of trying to draw any text on an image at all, I tried using [this](http://www.imagemagick.org/Usage/annotating/#annotating) example from the documentation (specifically the section that covers "Watermarking with Text") to no avail.  I'm not receiving any errors however nothing is being done to the either the input or output image.

**Comment by fmw42** [comment]
What was your exact command line for it all? What is your IM version?  Are you sure you got the EXIF data you want?

**Reply by Trademark8901** [reply]
I was about to edit my discussion with version info but I am on IM7 and I am currently just using sample text and I'm getting no changes to the image.

Below is my exact command line:
`cd C:\Users\%USERNAME%\Desktop\ImageMagick`
`magick IMG.jpg -font Arial -pointsize 20 -draw "gravity south  fill black  text 0,12  'Copyright'  fill white  text 1,11  'Copyright' " IMG2.jpg`

I have also tried the same command with "magick" & "magick convert".

**Reply by fmw42** [reply]
Works fine for me on IM 6.9.11.33 Q16 Mac OSX

`convert lena.jpg -font Arial -pointsize 20 -draw "gravity south fill black text 0,12 'Copyright' fill white text 1,11 'Copyright' " x.jpg`

Also IM 7.0.10.33

`magick lena.jpg -font Arial -pointsize 20 -draw "gravity south fill black text 0,12 'Copyright' fill white text 1,11 'Copyright' " x.jpg`

![x](https://user-images.githubusercontent.com/13458218/95271667-5043cc80-07f3-11eb-9605-f8f79fe37508.jpg)



On IM 7, you should be using magick not convert and not magick convert.  Also I do not suggest you run your commands from the imagmagick install directory if that is what you are doing.

What do you get from:

`magick -version`

or 

`magick.exe -version `

since on Windows?

**Reply by Trademark8901** [reply]
Version: ImageMagick 7.0.10-33 Q16 x64 2020-10-04 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Visual C++: 192729111
Features: Cipher DPC OpenCL
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib


I am using the portable version as I do not have administrative access on this computer, that's why I was running from the install directory.

**Reply by fmw42** [reply]
Do you have the arial font on your system and is it listed when you do 

magick -list font

If not, try using the full path to the font file itself.

If that does not work, then I will have to leave it to the Windows experts, since I do not use Windows. But apart from using convert (should be using magick), I do not see anything that is wrong syntax. Your install does not have fontconfig. But that should not matter especially if you use the full path the font file itself.

**Reply by Trademark8901** [reply]
I do have arial and it appears to be in that list.

How would I type the full font path? When I try `C:\Windows\Fonts\Arial.ttf`  after `-font` it reads all the backslashes as a single filename. (`C:WindowsFontsArial.ttf`)

**Reply by fmw42** [reply]
You are comparing the output to the input, correct?  Not just looking at the input after processing.  What is the size of your input (WxH)?

**Reply by Trademark8901** [reply]
Yes and it is generating what is essentially a copy-paste of the original image. The size of the image (which is just a sample image pulled from my desktop) is: 3456 x 4608

**Comment by snibgo** [comment, marked as answer]
Your command works fine for me, IM v7.0.8-64 on Windows 8.1.

> ... -pointsize 20 ...
The size of the image (which is just a sample image pulled from my desktop) is: 3456 x 4608

That is a very small pointsize on a large image. Have you looked carefully at bottom-centre of the result?

**Reply by Trademark8901** [reply]
You know, I thought about just this and kept checking the _bottom-centre_ but I couldn't see anything. So I just upped the point size and it turns out at one point I rotated the image and ImageMagick keeps the _original bottom-centre_ so it was on the right side of the image (and it was very very small).

**Reply by fmw42** [reply]
Are you running on Windows or Unix under windows?  If the former, are you sure that the "magick" portable install is compatible with Windows?

**Reply by fmw42** [reply]
Your JPG was likely taken at a different orientation. Add -auto-orient to your command line after reading the input. That should correct the orientation issue.
